ml-agents/mlagents/trainers/optimizer/torch_optimizer.py

```diff
@@ -85,18 +54,17 @@ def get_trajectory_value_estimates(
         else:
             visual_obs = []
 
-        memory = torch.zeros([1, len(vector_obs[0]), self.policy.m_size])
+        memory = torch.zeros([1, 1, self.policy.m_size])
 
         next_obs = np.concatenate(next_obs, axis=-1)
         next_obs = [ModelUtils.list_to_tensor(next_obs).unsqueeze(0)]
-        next_memory = torch.zeros([1, 1, self.policy.m_size])
 
-        value_estimates = self.policy.actor_critic.critic_pass(
-            vector_obs, visual_obs, memory
+        value_estimates, next_memory = self.policy.actor_critic.critic_pass(
+            vector_obs, visual_obs, memory, sequence_length=batch.num_experiences
         )
 
-        next_value_estimate = self.policy.actor_critic.critic_pass(
-            next_obs, next_obs, next_memory
+        next_value_estimate, _ = self.policy.actor_critic.critic_pass(
+            next_obs, next_obs, next_memory, sequence_length=1
```

Comment:
Does this work? it seems that critic_pass takes a vector obs and a visual obs but here we feed the same obs as both visual and vector.

Reply:
Welp, that's a bug 🙂

Reply:
Fixed